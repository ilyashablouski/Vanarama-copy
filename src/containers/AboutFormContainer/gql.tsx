import { useMutation, useQuery, gql } from '@apollo/client';
import {
  CreateUpdatePersonMutation,
  CreateUpdatePersonMutationVariables,
} from '../../../generated/CreateUpdatePersonMutation';
import {
  GetAboutYouDataQuery,
  GetAboutYouDataQueryVariables,
} from '../../../generated/GetAboutYouDataQuery';
import { GetAboutYouPageQuery } from '../../../generated/GetAboutYouPageQuery';
import AboutForm from '../../components/AboutForm';

export const CREATE_UPDATE_PERSON = gql`
  mutation CreateUpdatePersonMutation($input: PersonInputObject!) {
    createUpdatePerson(input: $input) {
      ...AboutFormPerson
    }
  }
  ${AboutForm.fragments.person}
`;

export const GET_ABOUT_YOU_DATA = gql`
  query GetAboutYouDataQuery($uuid: ID!) {
    personByUuid(uuid: $uuid) {
      ...AboutFormPerson
    }
  }
  ${AboutForm.fragments.person}
`;

export const GET_ABOUT_YOU_PAGE_DATA = gql`
  query GetAboutYouPageQuery {
    allDropDowns {
      ...AboutFormDropdownData
    }
  }
  ${AboutForm.fragments.dropdownData}
`;

export function useAboutPageDataQuery() {
  return useQuery<GetAboutYouPageQuery>(GET_ABOUT_YOU_PAGE_DATA, {
    fetchPolicy: 'no-cache',
  });
}

export function useAboutYouData(personByUuid?: string) {
  return useQuery<GetAboutYouDataQuery, GetAboutYouDataQueryVariables>(
    GET_ABOUT_YOU_DATA,
    {
      variables: {
        /**
         * If there's not a personUuid then we can set it to anything
         * as it will be excluded from the query anyway
         */
        uuid: personByUuid || '🐔',
      },
      fetchPolicy: 'no-cache',
      skip: !personByUuid,
    },
  );
}

export function useCreatePerson(
  onCompleted?: (data: CreateUpdatePersonMutation) => void,
) {
  return useMutation<
    CreateUpdatePersonMutation,
    CreateUpdatePersonMutationVariables
  >(CREATE_UPDATE_PERSON, {
    onCompleted,
  });
}
