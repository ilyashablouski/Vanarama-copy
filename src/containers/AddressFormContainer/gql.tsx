import { useMutation, useQuery } from '@apollo/react-hooks';
import { gql } from 'apollo-boost';
import {
  GetAddressContainerDataQuery as Query,
  GetAddressContainerDataQueryVariables as QueryVariables,
} from '../../../generated/GetAddressContainerDataQuery';
import {
  SaveAddressHistoryMutation as Mutation,
  SaveAddressHistoryMutationVariables as MutationVariables,
} from '../../../generated/SaveAddressHistoryMutation';
import AddressForm from '../../components/AddressForm/AddressForm';

export const GET_ADDRESS_CONTAINER_DATA = gql`
  query GetAddressContainerDataQuery($uuid: ID!) {
    personByUuid(uuid: $uuid) {
      uuid
      partyId
      addresses {
        ...AddressFormAddresses
      }
    }
    allDropDowns {
      ...AddressFormDropDownData
    }
  }
  ${AddressForm.fragments.addresses}
  ${AddressForm.fragments.dropDownData}
`;

export const SAVE_ADDRESS_HISTORY = gql`
  mutation SaveAddressHistoryMutation($input: AddressHistoryInputObject!) {
    createUpdateAddress(input: $input) {
      ...AddressFormAddresses
    }
  }
  ${AddressForm.fragments.addresses}
`;

export function useAddressData(personUuid: string) {
  return useQuery<Query, QueryVariables>(GET_ADDRESS_CONTAINER_DATA, {
    variables: {
      uuid: personUuid,
    },
  });
}

export function useUpdateAddresses(
  personUuid: string,
  onCompleted: (data: Mutation) => void,
) {
  return useMutation<Mutation, MutationVariables>(SAVE_ADDRESS_HISTORY, {
    onCompleted,
    update: (store, result) => {
      // Read the data from our cache for this query.
      const data = store.readQuery<Query, QueryVariables>({
        query: GET_ADDRESS_CONTAINER_DATA,
        variables: { uuid: personUuid },
      });

      // Add the addresses from the mutation to the end.
      if (data?.personByUuid) {
        data.personByUuid.addresses = result.data?.createUpdateAddress || [];

        // Write our data back to the cache.
        store.writeQuery<Query, QueryVariables>({
          query: GET_ADDRESS_CONTAINER_DATA,
          variables: { uuid: personUuid },
          data,
        });
      }
    },
  });
}
