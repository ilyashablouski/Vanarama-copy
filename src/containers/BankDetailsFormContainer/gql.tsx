import { useMutation, useQuery } from '@apollo/react-hooks';
import { gql } from 'apollo-boost';
import {
  CreateUpdateBankAccountMutation as Mutation,
  CreateUpdateBankAccountMutationVariables as MutationVariables,
} from '../../../generated/CreateUpdateBankAccountMutation';
import {
  GetBankDetailsPageDataQuery as Query,
  GetBankDetailsPageDataQueryVariables as QueryVariables,
} from '../../../generated/GetBankDetailsPageDataQuery';
import BankDetails from '../../components/BankDetails';

export const GET_BANK_DETAILS_PAGE_DATA = gql`
  query GetBankDetailsPageDataQuery($uuid: ID!) {
    personByUuid(uuid: $uuid) {
      uuid
      partyId
      bankAccounts {
        ...BankDetailsAccount
      }
    }
  }
  ${BankDetails.fragments.account}
`;

export const CREATE_UPDATE_BANK_ACCOUNT = gql`
  mutation CreateUpdateBankAccountMutation($input: BankAccountInputObject) {
    createUpdateBankAccount(input: $input) {
      ...BankDetailsAccount
    }
  }
  ${BankDetails.fragments.account}
`;

export function useBankDetails(personUuid: string) {
  return useQuery<Query, QueryVariables>(GET_BANK_DETAILS_PAGE_DATA, {
    variables: { uuid: personUuid },
  });
}

export function useUpdateBankDetails(
  personUuid: string,
  onCompleted: () => void,
) {
  return useMutation<Mutation, MutationVariables>(CREATE_UPDATE_BANK_ACCOUNT, {
    onCompleted,
    update: (store, result) => {
      // Read the data from our cache for this query.
      const data = store.readQuery<Query, QueryVariables>({
        query: GET_BANK_DETAILS_PAGE_DATA,
        variables: { uuid: personUuid },
      });

      // Update the person's bank details.
      if (data?.personByUuid) {
        data.personByUuid.bankAccounts = result.data?.createUpdateBankAccount
          ? [result.data?.createUpdateBankAccount]
          : null;

        // Write our data back to the cache.
        store.writeQuery<Query, QueryVariables>({
          query: GET_BANK_DETAILS_PAGE_DATA,
          variables: { uuid: personUuid },
          data,
        });
      }
    },
  });
}
