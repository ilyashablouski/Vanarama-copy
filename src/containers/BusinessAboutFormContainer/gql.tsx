import { gql, useMutation, useQuery, ApolloError } from '@apollo/client';
import { GetB2BAboutPageData } from '../../../generated/GetB2BAboutPageData';
import {
  SaveBusinessAboutYou,
  SaveBusinessAboutYouVariables,
} from '../../../generated/SaveBusinessAboutYou';
import BusinessAboutForm from '../../components/BusinessAboutForm/BusinessAboutForm';

export const GET_B2B_ABOUT_PAGE_DATA = gql`
  query GetB2BAboutPageData {
    allDropDowns {
      ...BusinessAboutFormDropDownData
    }
  }
  ${BusinessAboutForm.fragments.dropDownData}
`;

export const SAVE_BUSINESS_ABOUT_YOU = gql`
  mutation SaveBusinessAboutYou($input: PersonInputObject!) {
    createUpdateBusinessPerson(input: $input) {
      uuid
      companies {
        uuid
      }
    }
  }
`;

export function useAboutPageDataQuery() {
  return useQuery<GetB2BAboutPageData>(GET_B2B_ABOUT_PAGE_DATA);
}

export function useSaveAboutYouMutation(
  onCompleted?: (data: SaveBusinessAboutYou) => void,
  onError?: (error: ApolloError) => void,
) {
  return useMutation<SaveBusinessAboutYou, SaveBusinessAboutYouVariables>(
    SAVE_BUSINESS_ABOUT_YOU,
    { onCompleted, onError },
  );
}
