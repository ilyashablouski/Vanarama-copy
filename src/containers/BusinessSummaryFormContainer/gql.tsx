import { gql } from '@apollo/client';
import BusinessSummaryForm from '../../components/BusinessSummaryForm/BusinessSummaryForm';
import AboutForm from '../../components/AboutForm';

export const GET_COMPANY_SUMMARY = gql`
  query GetCompanySummaryQuery($uuid: ID!, $personUuid: ID!) {
    companyByUuid(uuid: $uuid) {
      ...SummaryFormCompany
    }
    personByUuid(uuid: $personUuid) {
      ...AboutFormPerson
    }
  }
  ${BusinessSummaryForm.fragments.company}
  ${AboutForm.fragments.person}
`;

export default GET_COMPANY_SUMMARY;
