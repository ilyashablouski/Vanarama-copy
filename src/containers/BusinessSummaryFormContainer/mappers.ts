import { GetCreditApplicationByOrderUuid_creditApplicationByOrderUuid as CreditApplication } from '../../../generated/GetCreditApplicationByOrderUuid';
import { VehicleTypeEnum } from '../../../generated/globalTypes';

export const mapCreditApplicationToCreditChecker = (
  data?: CreditApplication | null,
  partyId?: string | null,
) => ({
  partyId: partyId || '',
  orderUuid: data?.lineItem?.order?.uuid,
  vehicleType:
    data?.lineItem?.vehicleProduct?.vehicleType || VehicleTypeEnum.CAR,
  capId: data?.lineItem?.vehicleProduct?.derivativeCapId,
  creditApplicationUuid: data?.lineItem?.creditApplications?.[0]?.uuid || '',
  monthlyPayment: data?.lineItem?.vehicleProduct?.monthlyPayment || 0,
  depositPayment: data?.lineItem?.vehicleProduct?.depositPayment || 0,
});

export default mapCreditApplicationToCreditChecker;
