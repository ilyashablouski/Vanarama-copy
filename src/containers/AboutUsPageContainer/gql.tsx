import { gql, useQuery } from '@apollo/client';
import { GetAboutUsPageData as Query } from '../../../generated/GetAboutUsPageData';
import { IMAGE_FILE_FRAGMENT } from '../../gql/image';

export const GET_ABOUT_US_PAGE_DATA = gql`
  ${IMAGE_FILE_FRAGMENT}
  query GetAboutUsPageData($isPreview: Boolean) {
    aboutUsLandingPage(isPreview: $isPreview) {
      id
      body
      metaData {
        title
        name
        metaRobots
        metaDescription
        legacyUrl
        pageType
        canonicalUrl
        slug
        schema
        publishedOn
        breadcrumbs
      }
      featuredImage {
        file {
          ...imageFile
        }
      }
      sections {
        rowText {
          heading
          subHeading
          body
          rowTextLink {
            text
            url
          }
          link {
            text
            url
          }
        }
        cards {
          name
          cards {
            titleTag
            name
            title
            body
          }
        }
        carousel {
          name
          cards {
            titleTag
            name
            title
            body
          }
        }
      }
    }
  }
`;

export function useAboutUsPageData() {
  return useQuery<Query>(GET_ABOUT_US_PAGE_DATA);
}
