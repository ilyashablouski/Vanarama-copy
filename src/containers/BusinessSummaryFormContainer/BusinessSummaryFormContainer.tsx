import Loading from '@vanarama/uibook/lib/components/atoms/loading';
import React, { useEffect } from 'react';
import { useLazyQuery, ApolloError } from '@apollo/client';
import BusinessSummaryForm from '../../components/BusinessSummaryForm/BusinessSummaryForm';
import SoleTraderSummaryForm from '../../components/BusinessSummaryForm/SoleTraderSummaryForm';
import {
  GetCompanySummaryQuery,
  GetCompanySummaryQueryVariables,
  GetCompanySummaryQuery_personByUuid as PersonByUuid,
  GetCompanySummaryQuery_companyByUuid as CompanyByUuid,
} from '../../../generated/GetCompanySummaryQuery';
import { GET_COMPANY_SUMMARY, useUseFullCreditCheckerB2BMutation } from './gql';
import {
  useGetCreditApplicationByOrderUuid,
  useCreateUpdateCreditApplication,
} from '../../gql/creditApplication';
import { mapCreditApplicationToCreditChecker } from './mappers';
import { useGetPartyByUuidQuery } from '../../components/SummaryForm/gql';
import { GetPartyByUuid_partyByUuid as Party } from '../../../generated/GetPartyByUuid';
import { GetCreditApplicationByOrderUuid_creditApplicationByOrderUuid as CreditApplication } from '../../../generated/GetCreditApplicationByOrderUuid';

interface IProps {
  personUuid: string;
  companyUuid: string;
  orderId: string;
  isSoleTrader: boolean;
  onCompleted?: (emailAddresses: string | undefined) => void;
  onError?: (error: ApolloError) => void;
}

const BusinessSummaryFormContainer: React.FC<IProps> = ({
  companyUuid,
  orderId,
  personUuid,
  onCompleted,
  onError,
  isSoleTrader,
}) => {
  const [getDataSummary, getDataSummaryQueryOptions] = useLazyQuery<
    GetCompanySummaryQuery,
    GetCompanySummaryQueryVariables
  >(GET_COMPANY_SUMMARY, { fetchPolicy: 'no-cache' });
  const getCreditApplication = useGetCreditApplicationByOrderUuid(orderId);
  const [
    createUpdateCA,
    creditApplicationMutationOptions,
  ] = useCreateUpdateCreditApplication(orderId, () => {});
  const [
    submitFullCreditChecker,
    creditCheckerMutationOptions,
  ] = useUseFullCreditCheckerB2BMutation();
  const partyUuid =
    getCreditApplication.data?.creditApplicationByOrderUuid?.lineItem?.order
      ?.partyUuid;
  const getPartyByUuidQuery = useGetPartyByUuidQuery(partyUuid || '');

  const isSubmitting =
    creditApplicationMutationOptions.loading ||
    creditCheckerMutationOptions.loading;

  useEffect(() => {
    if (personUuid && companyUuid) {
      getDataSummary({
        variables: {
          uuid: companyUuid,
          personUuid,
        },
      });
    }
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [personUuid, companyUuid]);

  const error = getDataSummaryQueryOptions.error || getPartyByUuidQuery.error;
  if (error) {
    return <p>Error occurred: {error?.message}</p>;
  }

  if (
    getDataSummaryQueryOptions.loading ||
    getCreditApplication.loading ||
    getPartyByUuidQuery.loading ||
    (!getDataSummaryQueryOptions.data?.companyByUuid &&
      !getDataSummaryQueryOptions.data?.personByUuid)
  ) {
    return <Loading size="large" />;
  }

  const handleCreditApplicationSubmit = () =>
    createUpdateCA({
      variables: {
        input: {
          orderUuid: orderId,
          submittedAt: new Date(),
        },
      },
    });

  const handleCreditCheckerSubmit = (
    creditApplication?: CreditApplication | null,
    party?: Party | null,
  ) =>
    submitFullCreditChecker({
      variables: {
        input: mapCreditApplicationToCreditChecker(
          creditApplication,
          party?.company?.partyId || '',
        ),
      },
    });

  const handlePartyRefetch = (creditApplication?: CreditApplication | null) =>
    getPartyByUuidQuery.refetch({
      uuid: creditApplication?.lineItem?.order?.partyUuid || '',
    });

  const handleSubmit = () => {
    const personByUuid = getDataSummaryQueryOptions?.data?.personByUuid;
    handleCreditApplicationSubmit()
      .then(creditApplicationQuery =>
        handlePartyRefetch(
          creditApplicationQuery.data?.createUpdateCreditApplication,
        ).then(partyQuery =>
          handleCreditCheckerSubmit(
            creditApplicationQuery.data?.createUpdateCreditApplication,
            partyQuery.data?.partyByUuid,
          ),
        ),
      )
      .then(() => onCompleted?.(personByUuid?.emailAddresses[0].value))
      .catch(onError);
  };

  return (
    <>
      {isSoleTrader ? (
        <SoleTraderSummaryForm
          isSubmitting={isSubmitting}
          onSubmit={handleSubmit}
          creditApplication={
            getCreditApplication.data?.creditApplicationByOrderUuid
          }
          person={getDataSummaryQueryOptions.data.personByUuid as PersonByUuid}
          company={
            getDataSummaryQueryOptions.data.companyByUuid as CompanyByUuid
          }
        />
      ) : (
        <BusinessSummaryForm
          isSubmitting={isSubmitting}
          onSubmit={handleSubmit}
          creditApplication={
            getCreditApplication.data?.creditApplicationByOrderUuid
          }
          person={getDataSummaryQueryOptions.data.personByUuid as PersonByUuid}
          company={
            getDataSummaryQueryOptions.data.companyByUuid as CompanyByUuid
          }
        />
      )}
    </>
  );
};

export default BusinessSummaryFormContainer;
