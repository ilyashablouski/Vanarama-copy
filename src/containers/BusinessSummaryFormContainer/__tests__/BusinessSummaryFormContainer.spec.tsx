import React from 'react';
import { MockedProvider } from '@apollo/client/testing';
import renderer from 'react-test-renderer';
import BusinessSummaryFormContainer from '../BusinessSummaryFormContainer';
import { GetCompanySummaryQuery } from '../../../../generated/GetCompanySummaryQuery';
import { makeGetCreditApplicationMock } from '../../../gql/creditApplication';

// ARRANGE
const mockPush = jest.fn();
jest.mock('next/router', () => ({
  useRouter() {
    return {
      prefetch: () => null,
      push: mockPush,
      pathname: '/b2b/olaf/summary/[companyUuid]',
    };
  },
}));
jest.mock('../../../hooks/useImperativeQuery');

const mockParams = {
  personUuid: '895f87bc-14bf-407b-91c1-42ae3ebfb9e2',
  companyUuid: 'c4e8c130-8c71-4eeb-b2a8-b161426ef3a7',
  orderId: 'bf87b571-9846-49db-bd37-2c05c127efe8',
};

// const mocks = [
//   {
//     request: {
//       query: GET_COMPANY_SUMMARY,
//       variables: {
//         uuid: mockParams.companyUuid,
//         personUuid: mockParams.personUuid,
//       } as GetCompanySummaryQueryVariables,
//     },
//     result: {
//       data: {
//         companyByUuid: {
//           bankAccounts: [
//             {
//               __typename: 'BankAccountType',
//               uuid: 'c5939b96-2854-444b-9aec-52d634581eb0',
//               accountName: 'Eternal account',
//               accountNumber: '01234567',
//               joinedAt: '2007-08-01',
//               sortCode: '433335',
//               updatedAt: '2020-07-31T14:35:52.872+00:00',
//             },
//           ],
//           associates: [
//             {
//               uuid: '895f87bc-14bf-407b-91c1-42ae3ebfb9e2',
//               title: 'Miss',
//               firstName: 'Nastya',
//               lastName: 'Garbuz',
//               gender: 'Female',
//               dateOfBirth: '1999-02-03',
//               noOfDependants: 'None',
//               businessShare: 40,
//               roles: [
//                 {
//                   position: 'director',
//                 },
//               ],
//               addresses: [
//                 {
//                   serviceId: 'GB|RM|A|27452128|A1',
//                   propertyStatus: 'Owned with mortgage',
//                   startedOn: '2008-02-01',
//                   city: 'York',
//                   lineOne: 'Sky View',
//                   lineTwo: '5 Harrogate Road',
//                   postcode: 'YO51 9JD',
//                 },
//               ],
//             },
//           ],
//           uuid: 'c4e8c130-8c71-4eeb-b2a8-b161426ef3a7',
//           isVatRegistered: true,
//           tradesOutsideUk: true,
//           turnoverPercentageOutsideUk: [
//             {
//               country: 'Azerbaijan',
//               percentage: '12',
//             },
//             {
//               country: 'French Polynesia',
//               percentage: '34',
//             },
//           ],
//           vatNumber: '123456789',
//           __typename: 'CompanyType',
//           legalName: '12345 LIMITED',
//           companyNumber: '03799306',
//           companyNature: 'IT',
//           tradingSince: '1999-07-01',
//           addresses: [
//             {
//               serviceId: 'GB|RM|B|20230758|A1',
//               uuid: 'f3508691-e47d-4208-a637-b47a5eef7b02',
//               kind: 'registered',
//               lineOne: '4 Keymer Court',
//               lineTwo: '',
//               country: 'GB',
//               city: 'Burgess Hill',
//               postcode: 'RH15 0AA',
//             },
//           ],
//           emailAddresses: [
//             {
//               uuid: '0f7c0a3e-7930-4823-895b-07c79fd7f02f',
//               kind: 'Home',
//               value: 'a.harbuz121@reply.com',
//               primary: true,
//             },
//           ],
//           telephoneNumbers: [
//             {
//               uuid: 'e58c66c5-1bea-45e1-89f6-0af202d3660e',
//               kind: 'Mobile',
//               value: '00000000000',
//               primary: true,
//             },
//           ],
//         },
//         personByUuid: {
//           __typename: 'PersonType',
//           uuid: '895f87bc-14bf-407b-91c1-42ae3ebfb9e2',
//           title: 'Miss',
//           firstName: 'Nastya',
//           lastName: 'Garbuz',
//           emailAddresses: [
//             {
//               __typename: 'EmailAddressType',
//               uuid: '5ba1d32d-b704-441e-a3de-b4c4c8effc0c',
//               primary: true,
//               value: 'a.harbuz@reply.com',
//             },
//           ],
//           telephoneNumbers: [
//             {
//               __typename: 'TelephoneNumberType',
//               uuid: '6e007ebc-acfd-4bcc-8545-52e978980c9d',
//               kind: 'Mobile',
//               value: '09876543213',
//             },
//           ],
//           dateOfBirth: '1999-02-03',
//           countryOfBirth: 'Belarus',
//           nationality: 'Belarusian',
//           maritalStatus: 'Single',
//           noOfAdultsInHousehold: '1',
//           noOfDependants: 'None',
//           emailConsent: false,
//         },
//       } as GetCompanySummaryQuery,
//     },
//   } as MockedResponse,
// ];

const mockValue = {
  data: {
    companyByUuid: {
      bankAccounts: [
        {
          __typename: 'BankAccountType',
          uuid: 'c5939b96-2854-444b-9aec-52d634581eb0',
          accountName: 'Eternal account',
          accountNumber: '01234567',
          joinedAt: '2007-08-01',
          sortCode: '433335',
          updatedAt: '2020-07-31T14:35:52.872+00:00',
        },
      ],
      associates: [
        {
          uuid: '895f87bc-14bf-407b-91c1-42ae3ebfb9e2',
          title: 'Miss',
          firstName: 'Nastya',
          lastName: 'Garbuz',
          gender: 'Female',
          dateOfBirth: '1999-02-03',
          noOfDependants: 'None',
          businessShare: 40,
          roles: [
            {
              position: 'director',
            },
          ],
          addresses: [
            {
              serviceId: 'GB|RM|A|27452128|A1',
              propertyStatus: 'Owned with mortgage',
              startedOn: '2008-02-01',
              city: 'York',
              lineOne: 'Sky View',
              lineTwo: '5 Harrogate Road',
              postcode: 'YO51 9JD',
            },
          ],
        },
      ],
      uuid: 'c4e8c130-8c71-4eeb-b2a8-b161426ef3a7',
      isVatRegistered: true,
      tradesOutsideUk: true,
      turnoverPercentageOutsideUk: [
        {
          country: 'Azerbaijan',
          percentage: '12',
        },
        {
          country: 'French Polynesia',
          percentage: '34',
        },
      ],
      vatNumber: '123456789',
      __typename: 'CompanyType',
      legalName: '12345 LIMITED',
      companyNumber: '03799306',
      companyNature: 'IT',
      tradingSince: '1999-07-01',
      addresses: [
        {
          serviceId: 'GB|RM|B|20230758|A1',
          uuid: 'f3508691-e47d-4208-a637-b47a5eef7b02',
          kind: 'registered',
          lineOne: '4 Keymer Court',
          lineTwo: '',
          country: 'GB',
          city: 'Burgess Hill',
          postcode: 'RH15 0AA',
        },
      ],
      emailAddresses: [
        {
          uuid: '0f7c0a3e-7930-4823-895b-07c79fd7f02f',
          kind: 'Home',
          value: 'a.harbuz121@reply.com',
          primary: true,
        },
      ],
      telephoneNumbers: [
        {
          uuid: 'e58c66c5-1bea-45e1-89f6-0af202d3660e',
          kind: 'Mobile',
          value: '00000000000',
          primary: true,
        },
      ],
    },
    personByUuid: {
      __typename: 'PersonType',
      uuid: '895f87bc-14bf-407b-91c1-42ae3ebfb9e2',
      title: 'Miss',
      firstName: 'Nastya',
      lastName: 'Garbuz',
      emailAddresses: [
        {
          __typename: 'EmailAddressType',
          uuid: '5ba1d32d-b704-441e-a3de-b4c4c8effc0c',
          primary: true,
          value: 'a.harbuz@reply.com',
        },
      ],
      telephoneNumbers: [
        {
          __typename: 'TelephoneNumberType',
          uuid: '6e007ebc-acfd-4bcc-8545-52e978980c9d',
          kind: 'Mobile',
          value: '09876543213',
        },
      ],
      dateOfBirth: '1999-02-03',
      countryOfBirth: 'Belarus',
      nationality: 'Belarusian',
      maritalStatus: 'Single',
      noOfAdultsInHousehold: '1',
      noOfDependants: 'None',
      emailConsent: false,
    },
  } as GetCompanySummaryQuery,
};

describe('<BusinessSummaryFormContainer />', () => {
  it('should render correctly with data', () => {
    jest.mock('../../../hooks/useImperativeQuery', () => [
      jest.fn(),
      {
        loading: false,
        mockValue,
      },
    ]);

    const getComponent = () => {
      return renderer
        .create(
          <MockedProvider mocks={[makeGetCreditApplicationMock('')]}>
            <BusinessSummaryFormContainer {...mockParams} />
          </MockedProvider>,
        )
        .toJSON();
    };

    const tree = getComponent();
    expect(tree).toMatchSnapshot();
  });

  // xit('should redirect to the thank you page when clicking "Continue"', async () => {
  //   jest.mock('../../../hooks/useImperativeQuery', () => [
  //     jest.fn(),
  //     {
  //       loading: false,
  //       mockValue,
  //     },
  //   ]);

  //   const component = renderer.create(
  //     <BusinessSummaryFormContainer {...mockParams} />,
  //   );

  //   component.root
  //     .findByProps({
  //       'data-testid': 'olaf_summary_continue_buttton',
  //     })
  //     .props.onClick();

  //   // ASSERT
  //   expect(mockPush).toHaveBeenCalledTimes(1);
  //   expect(mockPush).toHaveBeenCalledWith('/olaf/thank-you/[orderId]');
  // });

  // xit('should redirect to about page when clicking "Edit" on the "Your Details" section', async () => {
  //   // ACT
  //   render(
  //     <MockedProvider addTypename={false} mocks={mocks}>
  //       <BusinessSummaryFormContainer {...mockParams} />
  //     </MockedProvider>,
  //   );

  //   // Wait for the data to load
  //   await screen.findByTestId('summary-heading');

  //   fireEvent.click(screen.getByTestId(/edit-your-details/));

  //   // ASSERT
  //   expect(mockPush).toHaveBeenCalledTimes(1);
  //   expect(mockPush).toHaveBeenCalledWith(
  //     '/olaf/about/[uuid]?redirect=summary',
  //     '/olaf/about/fd2333b8-6da1-47d2-837d-bc69849e0764?redirect=summary',
  //   );
  // });

  // xit('should redirect to employment history page when clicking "Edit" on the "Employment History" section', async () => {
  //   // ACT
  //   render(
  //     <MockedProvider addTypename={false} mocks={mocks}>
  //       <BusinessSummaryFormContainer {...mockParams} />
  //     </MockedProvider>,
  //   );

  //   // Wait for the data to load
  //   await screen.findByTestId('summary-heading');

  //   fireEvent.click(screen.getByTestId(/edit-employment-history/));

  //   // ASSERT
  //   expect(mockPush).toHaveBeenCalledTimes(1);
  //   expect(mockPush).toHaveBeenCalledWith(
  //     '/olaf/employment-history/[uuid]?redirect=summary',
  //     '/olaf/employment-history/fd2333b8-6da1-47d2-837d-bc69849e0764?redirect=summary',
  //   );
  // });

  // xit('should redirect to expenses page when clicking "Edit" on the "Monthly Income" section', async () => {
  //   // ACT
  //   render(
  //     <MockedProvider addTypename={false} mocks={mocks}>
  //       <BusinessSummaryFormContainer {...mockParams} />
  //     </MockedProvider>,
  //   );

  //   // Wait for the data to load
  //   await screen.findByTestId('summary-heading');

  //   fireEvent.click(screen.getByTestId(/edit-expenses/));

  //   // ASSERT
  //   expect(mockPush).toHaveBeenCalledTimes(1);
  //   expect(mockPush).toHaveBeenCalledWith(
  //     '/olaf/expenses/[uuid]?redirect=summary',
  //     '/olaf/expenses/fd2333b8-6da1-47d2-837d-bc69849e0764?redirect=summary',
  //   );
  // });

  // xit('should redirect to bank details page when clicking "Edit" on the "Bank Details" section', async () => {
  //   // ACT
  //   render(
  //     <MockedProvider addTypename={false} mocks={mocks}>
  //       <BusinessSummaryFormContainer {...mockParams} />
  //     </MockedProvider>,
  //   );

  //   // Wait for the data to load
  //   await screen.findByTestId('summary-heading');

  //   fireEvent.click(screen.getByTestId(/edit-bank-details/));

  //   // ASSERT
  //   expect(mockPush).toHaveBeenCalledTimes(1);
  //   expect(mockPush).toHaveBeenCalledWith(
  //     '/olaf/bank-details/[uuid]?redirect=summary',
  //     '/olaf/bank-details/fd2333b8-6da1-47d2-837d-bc69849e0764?redirect=summary',
  //   );
  // });
});
