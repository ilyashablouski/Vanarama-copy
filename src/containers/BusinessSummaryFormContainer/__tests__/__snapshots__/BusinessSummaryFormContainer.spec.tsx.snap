// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BusinessSummaryFormContainer /> should render correctly with data 1`] = `
<div
  className="loading -large"
>
  <svg
    className="loading--svg"
    viewBox="-75 -75 150 150"
  >
    <circle
      className="loading--stroke"
      cx="0"
      cy="0"
      r="37.5"
    />
  </svg>
</div>
`;
